### PR TITLE
Nc news 4

### PR DESCRIPTION
**Pull request by gilbertouk**
implemented task 4

### CODE REVIEW

File: __tests__/api/app.test.js [hunk 1]
```diff
@@ -66,4 +65,54 @@ describe('Testing app', () => {
...
+              article: {
+                article_id: 1,
+                title: 'Living in the shadow of a great man',
+                topic: 'mitch',
+                author: 'butter_bridge',
+                body: 'I find this existence challenging',
+                created_at: '2020-07-09T20:11:00.000Z',
+                votes: 100,
+                article_img_url:
+                  'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/api/app.test.js]: Really good test, brilliant that you're asserting the exact article you expect the endpoint to return when given an id of 1

File: models/articles.models.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+const selectArticleById = (article_id) => {
+  return db
+    .query('SELECT * FROM articles WHERE article_id = $1', [article_id])
+    .then(({ rows }) => {
+      if (rows.length === 0) {
+        return Promise.reject({ status: 404, msg: 'Resource not found' });
+      }
+
+      return rows[0];
+    });
```
Comment (Pipe-Smoking-Rabbit) [inline models/articles.models.js:12]: Brilliant!

File: app.js [hunk 3]
```diff
@@ -1,13 +1,32 @@
...
+  }
+});
+
+app.use((err, req, res, next) => {
+  if (err.code === '22P02') {
+    res.status(400).send({ msg: 'Bad request' });
+  } else {
+    next(err);
+  }
+});
```
Comment (Pipe-Smoking-Rabbit) [inline app.js:29]: Great error handling

File: __tests__/api/app.test.js [hunk 4]
```diff
@@ -66,4 +65,54 @@ describe('Testing app', () => {
...
+      });
+
+      test('GET: 400 status when given invalid article_id', () => {
+        return request(app)
+          .get('/api/articles/invalid_id')
+          .expect(400)
+          .then(({ body }) => {
+            expect(body.msg).toBe('Bad request');
+          });
+      });
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/api/app.test.js:113]: Excellent error testing

File: __tests__/api/app.test.js [hunk 5]
```diff
@@ -66,4 +65,54 @@ describe('Testing app', () => {
...
+
+      test('GET: 200 status responds with an article object data', () => {
+        return request(app)
+          .get('/api/articles/1')
+          .then(({ body }) => {
+            const article = body;
+
+            const expectArticle = {
+              article: {
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/api/app.test.js]: slight issue with deconstruction here,  if you change line 79 to be `const { article } = body;` then you won't need to have the key of _article_ on line 82

**Review by Pipe-Smoking-Rabbit — APPROVED** [review]
Really good stuff on this endpoint! 👍 

I have left a comment regarding the deconstruction in the test as it's possible to save a line by deconstructing `article`

File: __tests__/api/app.test.js (hunk 5 above)
Comment (gilbertouk) [reply __tests__/api/app.test.js]: Done!

**Pull request merged**